### ISSUE TITLE
Pallas YAT attention does not mask ragged K/V tiles or bound causal cross-attention

### ISSUE THREAD

**Issue by mlnomadpy** [azettaai/nmn#49]
## Summary

`pallas_yat_l1_attention` silently produces incorrect forward outputs when `kv_len` is not divisible by `block_k`. Causal cross-attention also reads beyond the available K/V blocks when `q_len > kv_len`.

The public docstring says the tile sizes must divide the sequence lengths, but the implementation uses `pl.cdiv` and does not validate that precondition. Out-of-bounds/padded tile elements are not masked from scores and normalization. The causal loop bound is also not capped by the available number of K/V tiles.

## Affected code

- `src/nmn/nnx/layers/attention/pallas_yat_attention.py`
- Forward and backward tiled loops

## Reproduction

On commit `4f0298d`, CPU Pallas interpret mode:

```python
import jax
import jax.numpy as jnp
from nmn.nnx.layers.attention.pallas_yat_attention import pallas_yat_l1_attention
from nmn.nnx.layers.attention.fused_yat_attention import fused_yat_l1_attention

def compare(q_len, kv_len, causal=False):
    q = jax.random.normal(jax.random.key(0), (1, q_len, 2, 4))
    k = jax.random.normal(jax.random.key(1), (1, kv_len, 2, 4))
    v = jax.random.normal(jax.random.key(2), (1, kv_len, 2, 4))
    mask = None
    if causal:
        mask = jnp.arange(q_len)[:, None] >= jnp.arange(kv_len)[None, :]
    out = pallas_yat_l1_attention(
        q, k, v, causal=causal, block_q=4, block_k=4, interpret=True
    )
    ref = fused_yat_l1_attention(q, k, v, mask=mask)
    return float(jnp.max(jnp.abs(out - ref)))

print(compare(8, 10, causal=False))
print(compare(12, 8, causal=True))
```

Observed:

```
ragged kv_len: 0.365541 max error
causal q_len > kv_len: 0.267788 max error
```

Aligned self-attention in the same setup matches within approximately `2.4e-7`.

## Impact

Variable-length batches and causal cross-attention can silently return wrong values on the Pallas GPU/TPU path.

## Acceptance criteria

- [ ] Partial K/V and Q tiles are masked in forward and backward, or unsupported shapes are rejected before kernel launch.
- [ ] Causal loop bounds never exceed the available K/V blocks.
- [ ] Forward and gradient parity tests cover ragged Q, ragged K/V, `q_len < kv_len`, and `q_len > kv_len`.
- [ ] Tests run in `interpret=True` and on a supported GPU or TPU backend.

**Comment by mlnomadpy** [comment]
PR #83 fixed ragged/cross-causal bounds and passed independent CPU Pallas `interpret=True` parity review, but the issue acceptance criterion also requires execution on a supported GPU or TPU backend. That hardware validation has not been run locally, so this issue is reopened until accelerator lowering/execution is verified.

**Comment by mlnomadpy** [comment]
Fixed by merged PR #100 (merge f60a4d9f). Native Kaggle TPU v5e validation at exact head 5c9d25b covered ragged q<kv, ragged q>kv, causal/noncausal, multi-tile, multi-batch, and unequal value dimensions, including forward and dQ/dK/dV parity. Worst forward max-abs error was 4.77e-7; worst gradient max-abs error was 9.35e-5; all gradient cosines were effectively 1.0. Validation: https://www.kaggle.com/code/skywolfmo/nmn-pallas-gpu-validation-issues-48-49
